authorizations/authorization-ranger/src/main/java/org/apache/gravitino/authorization/ranger/RangerAuthorizationHDFSPlugin.java

```diff
@@ -740,4 +776,39 @@ protected Map<String, String> getServiceConfigs(Map<String, String> config) {
                 RangerAuthorizationProperties.FS_DEFAULT_VALUE))
         .build();
   }
+
+  private interface PathExtractor {
+    String getPath(Entity.EntityType type, String location);
+  }
+
+  private static class TableOrFilesetPathExtractor implements PathExtractor {
+    @Override
+    public String getPath(Entity.EntityType type, String location) {
+      if (type == Entity.EntityType.CATALOG) {
+        return String.format("%s/*/*/", location);
+      } else if (type == Entity.EntityType.SCHEMA) {
+        return String.format("%s/*/", location);
+      }
+      if (type == Entity.EntityType.TABLE || type == Entity.EntityType.FILESET) {
+        return location;
+      } else {
+        throw new AuthorizationPluginException(
+            "Don't support %s to get table or fileset location", type);
+      }
+    }
+  }
+
+  private static class SchemaPathExtractor implements PathExtractor {
+    @Override
+    public String getPath(Entity.EntityType type, String location) {
+      if (type == Entity.EntityType.CATALOG) {
+        return String.format("%s/*/", location);
+      } else if (type == Entity.EntityType.SCHEMA) {
+        return location;
+      } else {
+        throw new AuthorizationPluginException(
+            "Don't support %s to get table or fileset location", type);
```

Comment:
It's not allowed/supported to extract table or fileset path from entity %s

Reply:
Done.